### DISCUSSION TITLE
我的dev内核与tun内核在更新后莫名消失了，手动安装内核与重新安装clash都无法出现

### DISCUSSION THREAD

**Discussion by qing389** [vernesong/OpenClash#4080] [Q&A] [answered]
![微信图片_20241002114939](https://github.com/user-attachments/assets/f2b197f6-de8d-4607-8c41-bbda58f28b44)
![微信图片_20241002115013](https://github.com/user-attachments/assets/8fc6aa50-3210-41e3-8352-7a8fce26089d)
正常应该有dev tun mata三种内核，我只剩下一种了，并且我手动安装也无法复原

**Comment by qing389** [comment]
图一是我的openclash，图二是教程中的openclash，重装也不起效

**Reply by 19901019** [reply, marked as answer]
0.46.033开始只有meta了，没有dev和tun了

**Reply by qing389** [reply]
谢谢您的解答，请问我如果想要dev和tun只能重装老版本clash了吗？

**Reply by 19901019** [reply]
是的，0.46.014以下版本才有。
